feature/appdetail/src/test/kotlin/com/merxury/blocker/feature/appdetail/AppDetailViewModelTest.kt

```diff
@@ -240,10 +242,240 @@ class AppDetailViewModelTest {
 
     @Test
     fun tabState_whenSwitchTab_thenUpdateSelectedItem() = runTest {
-        val collectJob1 = launch(UnconfinedTestDispatcher()) { viewModel.tabState.collect() }
+        val collectJob = launch(UnconfinedTestDispatcher()) { viewModel.tabState.collect() }
         viewModel.switchTab(AppDetailTabs.Receiver)
         assertEquals(AppDetailTabs.Receiver, viewModel.tabState.value.selectedItem)
+        collectJob.cancel()
+    }
+
+    @Test
+    fun appBarUiState_whenSwitchTab_thenUpdateActionButtons() = runTest {
+        val collectJob = launch(UnconfinedTestDispatcher()) { viewModel.appBarUiState.collect() }
+        viewModel.switchTab(AppDetailTabs.Receiver)
+        assertEquals(
+            AppBarUiState(
+                actions = listOf(
+                    SEARCH,
+                    MORE,
+                ),
+            ),
+            viewModel.appBarUiState.value,
+        )
+        collectJob.cancel()
+    }
+
+    @Test
+    fun appBarUiState_whenSearchNotResult_thenSwitchToInfoTab() = runTest {
+        val collectJob1 = launch(UnconfinedTestDispatcher()) { viewModel.appBarUiState.collect() }
+        val collectJob2 = launch(UnconfinedTestDispatcher()) { viewModel.tabState.collect() }
+        viewModel.switchTab(AppDetailTabs.Receiver)
+        viewModel.search("123")
+        viewModel.changeSearchMode(true)
+        viewModel.loadTabInfo()
+        assertEquals(
+            AppBarUiState(
+                isSearchMode = true,
+                keyword = "123",
+            ),
+            viewModel.appBarUiState.value,
+        )
+        assertEquals(AppDetailTabs.Info, viewModel.tabState.value.selectedItem)
         collectJob1.cancel()
+        collectJob2.cancel()
+    }
+
+    @Test
+    fun appBarUiState_whenSearchResult_thenShowSearchResult() = runTest {
+        val collectJob = launch(UnconfinedTestDispatcher()) { viewModel.appInfoUiState.collect() }
+        appRepository.sendAppList(sampleAppList)
+        userDataRepository.sendUserData(defaultUserData)
+        viewModel.loadAppInfo()
+        componentRepository.sendComponentList(sampleComponentList)
+        componentDetailRepository.sendComponentDetail(sampleComponentDetailList)
+        viewModel.loadComponentList()
+        viewModel.changeSearchMode(true)
+        val keyword = "Activity"
+        viewModel.search(keyword)
+        assertEquals(
+            Result.Success(
+                ComponentSearchResult(
+                    app = sampleAppList.first().toAppItem(),
+                    activity = sampleComponentList.filter {
+                        it.type == ACTIVITY && it.name.contains(
```

Comment:
You don't need to filter by type since it only filters the name.

Reply:
It's because I need to fill the data list into ComponentSearchResult